### PR TITLE
Adding methods for selecting and confirming allergies on AllergyPage.java

### PR DESCRIPTION
**Pull request by Ndacyayisenga-droid**
cc @dkayiwa  @kdaud  @sherrif10

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
     private static final By ALLERGY_STATUS = By.cssSelector(".allergyStatus");
     private static final By ADD_NO_KNOWN = By.xpath("//*[contains(text(),'No Known Allergy')]");
     private static final By DELETE_NO_KNOWN = By.className("delete-action");
+    private static final By SELECT_NEW_ALLERGY = By.id("allergen-73667-label");  //codein
+    private static final By CONFIRM_SAVE_ALLERGY_BUTTON = By.id("addAllergyBtn");
```
Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java:28]: Which kind of Id is that

Comment (Ndacyayisenga-droid) [reply ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java:28]: > Which kind of Id is that

@sherrif10 can you try to inspect those `buttons` and check their ids

**Comment by kdaud** [comment]
@Ndacyayisenga-droid are you sure your branch is up to date with master?

File: ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java (hunk 1 above)
Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java:28]: True thanks , Can you make it    **SAVE_ALLERGY_BUTTON**

Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java:28]: Probably we can also add Cancel Button

**Comment by Ndacyayisenga-droid** [comment]
@kdaud  @sherrif10  are we  at peace now?

**Comment by kdaud** [comment]
@Ndacyayisenga-droid did you take a look at what [allergy-page](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L32-L48) can accomplish for you?

**Comment by Ndacyayisenga-droid** [comment]
> @Ndacyayisenga-droid did you take a look at what these [methods](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L32-L48) can accomplish for you?

@kdaud could there be some duplicates am not noticing?

**Comment by kdaud** [comment]
@Ndacyayisenga-droid what do you think of the [clickOnNewAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L32-L35) and your new method [selectNewAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/4c805285b15ef79806496801e2b84cf3bde5a118/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L78-L80)?

what do you think of [clickOnSaveAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AddOrEditAllergyPage.java#L37-L40) and your new method [confirmSaveAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/4c805285b15ef79806496801e2b84cf3bde5a118/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L82-L84)?

**Comment by sherrif10** [comment]
Sure Refer to @kdaud comment please , thanks alot

**Comment by Ndacyayisenga-droid** [comment]
> > could there be some duplicates am not noticing?
> 
> @Ndacyayisenga-droid what do you think of the method [clickOnNewAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L32-L35) and your new method [selectNewAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/4c805285b15ef79806496801e2b84cf3bde5a118/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L78-L80)?
> 
> what about the method [clickOnSaveAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AddOrEditAllergyPage.java#L37-L40) and your new method [confirmSaveAllergy()](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/4c805285b15ef79806496801e2b84cf3bde5a118/ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java#L82-L84)?

@kdaud am tempted to think `clickOnNewAllergy()` and `selectNewAllergy()` are different sorry about `clickOnSaveAllergy` and `confirmSaveAllergy`

**Comment by kdaud** [comment]
> am tempted to think `clickOnNewAllergy()` and `selectNewAllergy()` are different

@Ndacyayisenga-droid step back to look into what each method is accomplishing and find out whether there a difference!

**Comment by kdaud** [comment]
> am tempted to think `clickOnNewAllergy()` and `selectNewAllergy()` are different

@Ndacyayisenga-droid step back to look into what each method is accomplishing and find out whether there a difference!

**Comment by Ndacyayisenga-droid** [comment]
> > am tempted to think `clickOnNewAllergy()` and `selectNewAllergy()` are different
> 
> @Ndacyayisenga-droid step back to look at what each method is accomplishing and find out whether there a difference!

@kdaud after checking I found out `clickOnNewAllergy()` and `selectNewAllergy()` are different in my case

**Comment by kdaud** [comment]
@Ndacyayisenga-droid what does each method accomplish when invoked?

**Comment by Ndacyayisenga-droid** [comment]
@kdaud I think you mistyped the method `clickOnAddNewAllergy()` for `clickOnNewAllergy()` besides that is `selectNewAllergy()` selects an allergy ie codeine, asprini among others whereas `clickOnAddNewAllergy` just clicks **Add New Allergy** button

**Comment by kdaud** [comment]
@Ndacyayisenga-droid why not invoke [enterDrug(String drug)](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/82fa6e90ca4e7014f9ad0a564b1d96abbb345c2a/ui-tests/src/test/java/org/openmrs/reference/page/AddOrEditAllergyPage.java#L42) and pass the drug name in the method param?

File: ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java [hunk 2]
```diff
@@ -71,4 +72,8 @@ public void removeNoKnownAllergy() {
     public String getPageUrl() {
         return "/allergyui/allergies.page";
     }
+
+    public void confirmCancelAllergy(){
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java:76]: @Ndacyayisenga-droid is `confirmCancelAllergy()` a requirement for [RATEST-225](https://issues.openmrs.org/browse/RATEST-225)?

Comment (Ndacyayisenga-droid) [reply ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java]: cc @kdaud  @sherrif10

**Comment by sherrif10** [comment]
Check with travis failure

**Comment by sherrif10** [comment]
Check with travis failure

**Comment by Ndacyayisenga-droid** [comment]
> Check with travis failure

@sherrif10 I think someone  has to re-run this....
Can I now have a review on this on https://github.com/openmrs/openmrs-contrib-qaframework/pull/137 because it nolonger depends on this

**Comment by sherrif10** [comment]
Please push an empty  commit for this commit to be re run again

**Comment by kdaud** [comment]
@Ndacyayisenga-droid is this commit still necessary?

**Comment by Ndacyayisenga-droid** [comment]
@kdaud I dont think so

**Comment by kdaud** [comment]
> I dont think so

Better to close it then!

**Comment by Ndacyayisenga-droid** [comment]
> > I dont think so
> 
> Better to close it then!

Could you re-run https://github.com/openmrs/openmrs-contrib-qaframework/pull/137 then!

**Pull request closed without merging**